lib/awful/rules.lua

```diff
@@ -342,10 +342,41 @@ end
 
 rules.add_rule_source("awful.rules", apply_awful_rules, {"awful.spawn"}, {})
 
+local blacklisted_snid = {}
+
+--- Make startup notification work for some clients like XTerm. This is ugly
+-- but works often enough to be useful.
+local function get_startup_id_fallback(c)
+    if not c.pid then return end
+
+    local f = io.open("/proc/"..c.pid.."/environ", "rb")
+
+    -- It will only work on Linux, that's already 99% of the userbase.
+    if not f then return end
```

Comment:
For at least FreeBSD there seems to be `procstat`: https://forums.freebsd.org/threads/environment-variables-of-a-running-process.49190/